src/languageservice/parser/jsonParser07.ts

```diff
@@ -876,7 +876,7 @@ function validate(
       const val = getNodeValue(node);
       let enumValueMatch = false;
       for (const e of schema.enum) {
-        if (val === e || isAutoCompleteEqualMaybe(callFromAutoComplete, node, val, e)) {
+        if (equals(val, schema.enum, node.type) || isAutoCompleteEqualMaybe(callFromAutoComplete, node, val, e)) {
```

Comment:
I think this is wrong and should be:

```suggestion
        if (equals(val, e, node.type) || isAutoCompleteEqualMaybe(callFromAutoComplete, node, val, e)) {
```